core/src/main/java/hudson/tools/ZipExtractionInstaller.java

```diff
@@ -90,17 +95,25 @@ public FilePath performInstallation(ToolInstallation tool, Node node, TaskListen
         }
     }
 
+    /**
+     * Checks if the specified expected location already contains the installed version of the tool.
+     *
+     * This check needs to run fairly efficiently. The current implementation uses the URL of the archive
+     * based on the assumption that released bits do not change its content.
+     */
+    protected boolean isUpToDate(FilePath expectedLocation, String url) throws IOException, InterruptedException {
+        FilePath marker = expectedLocation.child(".installedFrom");
+        return marker.exists() && marker.readToString().equals(url);
+    }
+
     @Extension @Symbol("zip")
     public static class DescriptorImpl extends ToolInstallerDescriptor<ZipExtractionInstaller> {
 
         public String getDisplayName() {
             return Messages.ZipExtractionInstaller_DescriptorImpl_displayName();
         }
 
-        @RequirePOST
         public FormValidation doCheckUrl(@QueryParameter String value) {
-            Jenkins.getInstance().checkPermission(Jenkins.ADMINISTER);
```

Comment:
this is a bad idea (CSRF risk, DDoS)

Reply:
Not sure how this line ended up getting pulled out (explains the failing UT), guess when I sync'd upstream it didn't get pulled in right.